manifests/init.pp

```diff
@@ -60,6 +60,7 @@
   $install_dir           = $::puppet_agent::params::install_dir,
   $disable_proxy         = false,
   $install_options       = $::puppet_agent::params::install_options,
+  $skip_if_unavailable   = 'absent',
```

Comment:
Maybe we should have a yumrepo hash, that lets you override any particular parameters to setting up the repo... although that also sounds terrible.